### PR TITLE
 STM32WB Series support

### PR DESCRIPTION
**Pull request by erwango**
Add support for dual core, multi protocol wireless STM32 controllers.
Provide support for nucleo_wb55rg BLE target

### CODE REVIEW

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+  - ST morpho extension pin headers for full access to all STM32 I/Os
+
+- Integrated PCB antenna or footprint for SMA connector
+- On-board ST-LINK/V2-1 debugger/programmer with SWD connector
+- Flexible power-supply options: ST-LINK USB VBUS or external sources
+- On-board socket for CR2032 battery
+- On-board ST-LINK/V2-1 debugger/programmer with USB re- enumeration capability:
+  mass storage, virtual COM port and debug port
+
+.. image:: img/nucleowb55rg.jpg
```
Comment (dbkinder) [inline boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: The board image should be in ``boards/arm/nucleo_wb55rg/doc/img/nucleowb55rg.jpg`` and you've got it in ``boards/arm/nucleo_wb55rg/img/nucleowb55rg.jpg``

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: nucleo_wb55rg
+   :maybe-skip-config:
+   :goals: debug
+
+.. _Nucleo WB55RG website:
+   https://www.st.com/en/evaluation-tools/p-nucleo-wb55.html
+
+.. _STM32WB55RGG on www.st.com:
```
Comment (dbkinder) [inline boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: Looks like you've got an extra ``G`` on this label (hence the doc build error)

**Review by dbkinder — CHANGES_REQUESTED** [review]
Looks good, other than the two doc build errors...

**Review by dbkinder — APPROVED** [review]
+1 for docs

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+   :height: 339px
+   :alt: Nucleo WB55RG
+
+More information about the board can be found at the `Nucleo WB55RG website`_.
+
+Hardware
+********
+
+STM32WB55RG is an ultra-low-power dual core Arm Cortex-M4 MCU 64 MHz,Cortex-M0 32MHz
+with 1 Mbyte of Flash memory, Bluetooth 5, 802.15.4, USB, LCD, AES-259 SoC and
```
Comment (therealprof) [inline boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]:
```suggestion
with 1 Mbyte of Flash memory, Bluetooth 5, 802.15.4, USB, LCD, AES-256 SoC and
```

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+ - CRC calculation unit
+ - 96-bit unique ID
+ - 64-bit unique ID. Possibility to derive 802.15.5 64-bit and
+   Bluetooth|reg| Low Energy 48-bit EUI
+
+- Up to 72 fast I/Os, 70 of them 5 V-tolerant
+- Development support: serial wire debug (SWD), JTAG, Embedded Trace Macrocell |trade|
+
+
+More information about STM32L496ZG can be found here:
```
Comment (therealprof) [inline boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: STM32L496ZG?

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst (hunk 3 above)
Comment (erwango) [reply boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: @therealprof, txs for the catch. And while you're here, would you mind check my last comment on our pending discussion (https://github.com/zephyrproject-rtos/zephyr/pull/9300#discussion_r239447973)? Would be nice, txs.

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+
+- `STM32WB55RG on www.st.com`_
+- `STM32WB5RG datasheet`_
+
+Supported Features
+==================
+
+The Zephyr nucleo_l496zg board configuration supports the following hardware features:
```
Comment (therealprof) [inline boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: nucleo_l496zg?

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+Flashing
+========
+
+Nucleo WB55RG board includes an ST-LINK/V2-1 embedded debug tool
+interface.  This interface is not yet supported by the openocd version
+included in the Zephyr SDK. You can flash your application with drag and drop
+in the drive mounted when plugging your nucleo board to your PC.
+
+
+Flashing an application to Nucleo L496ZG
```
Comment (therealprof) [inline boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: dito

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst (hunk 3 above)
Comment (erwango) [reply boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: fixed

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst (hunk 6 above)
Comment (erwango) [reply boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: dito

File: dts/arm/st/wb/stm32wb.dtsi [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0: flash@8000000 {
+				compatible = "soc-nv-flash";
+				label = "FLASH_STM32";
+			};
+		};
+
```
Comment (galak) [inline dts/arm/st/wb/stm32wb.dtsi]: nit remove blank line.

**Review by galak — CHANGES_REQUESTED** [review]
One minor nit, otherwise looks good.

File: dts/arm/st/wb/stm32wb.dtsi (hunk 7 above)
Comment (erwango) [reply dts/arm/st/wb/stm32wb.dtsi]: Txs! Fixed now!

**Review by galak — APPROVED** [review]

**Pull request merged**